lib/typecheck.ts

```diff
@@ -368,6 +383,23 @@ export default class TypeChecker {
             throw new TypeError(`Invalid principal ${principal}, must be a contact or a group`);
     }
 
+    private _expandTypeScope(typeScope : TypeScope, key : string) : TypeScope[] {
```

Comment:
Can you add a comment explaining what this function does, and how it's used in _resolveOverload?